src/test/java/test/failedreporter/FailedReporterTest.java

```diff
@@ -31,30 +42,106 @@ public void tearDown() {
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">", 0);
+    substitutions = new String[] {"newTest2", "baseBeforeTest","baseAfterClass","baseBeforeClass","baseBeforeMethod"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
+  }
+
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigsInvolvingGroupsAtTestLevel() {
+    triggerTest(GroupsSampleBase.class.getPackage().getName(),true);
+    String[] substitutions = new String[] {GroupsPassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">",0);
+    substitutions = new String[] {"baseBeforeTest", "baseBeforeClassAlwaysRun","newTest2"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
+  }
+
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigsInvolvingGroupsAtSuiteLevel() {
+    triggerTest(GroupsSampleBase.class.getPackage().getName(),false);
+    String[] substitutions = new String[] {GroupsPassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">",0);
+    substitutions = new String[] {"baseBeforeTest", "baseBeforeClassAlwaysRun","newTest2"};
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>",1);
+  }
+
+  private void triggerTest(String packageName, boolean applyGroupSelectionAtTest) {
     TestNG tng = new TestNG();
     tng.setVerbose(0);
-    tng.setTestClasses(new Class[] { cls });
+    final XmlSuite suite = new XmlSuite();
+    suite.setName("1297_suite");
+
+    final XmlPackage xmlPackage = new XmlPackage();
+    xmlPackage.setName(packageName);
+    final XmlTest xmlTest = new XmlTest(suite);
+    xmlTest.setName("1297_test");
+    if (applyGroupSelectionAtTest) {
+      xmlTest.setGroups(getGroup());
+    } else {
+      suite.setGroups(getGroup());
+    }
+      xmlTest.setPackages(new ArrayList<XmlPackage>(){{add(xmlPackage);}});
+    suite.setTests(new ArrayList<XmlTest>(){{add(xmlTest);}});
+    tng.setXmlSuites(new ArrayList<XmlSuite>(){{add(suite);}});
     tng.setOutputDirectory(mTempDirectory.getAbsolutePath());
     tng.run();
+  }
+
+  private XmlGroups getGroup() {
+    XmlGroups xmlGroups = new XmlGroups();
+    XmlRun xmlRun = new XmlRun();
+    xmlRun.onInclude("run");
+    xmlGroups.setRun(xmlRun);
+    return xmlGroups;
+  }
+
+  private void testFailedReporter(String[] expectedMethods, String expectedLine, Class<?>... cls) {
+    triggerTest(cls);
     runAssertions(mTempDirectory,expectedMethods,expectedLine);
   }
 
-  public static void runAssertions(File outputDir, String[] expectedMethods, String expectedLine) {
+  private void triggerTest(Class<?>... cls) {
+    TestNG tng = new TestNG();
+    tng.setVerbose(0);
+    tng.setTestClasses(cls);
+    tng.setOutputDirectory(mTempDirectory.getAbsolutePath());
+    tng.run();
+  }
+
+  static void runAssertions(File outputDir, String[] expectedMethods, String expectedLine) {
```

Comment:
We cannot make this method private because it is being shared by `test.failedreporter.FailedReporterScenariosTest#runAssertions`